File: GildedRose/src/com/gildedrose/GildedRose.java
package com.gildedrose;

import com.gildedrose.items.*;

public class GildedRose {
    Item[] items;

    public GildedRose(Item[] items) {
        this.items = items;
    }

    public void updateQuality() {
        for (int i = 0; i < items.length; i++) {
             ItemDecorator decoratedItem = ItemDecoratorFactory.getInstance(items[i]);
             decoratedItem.update();
        }
    }
}

File: GildedRose.Net/GildedRose.cs
using System.Collections.Generic;

using GildedRose.Net.Items;

namespace GildedRose.Net
{
    class GildedRose
    {
        IList<Item> Items;
        public GildedRose(IList<Item> Items) 
        {
            this.Items = Items;
        }
        
        public void UpdateQuality()
        {
            for (var i = 0; i < Items.Count; i++)
            {
                ItemDecorator decoratedItem = ItemDecoratorFactory.CreateInstance(Items[i]);
                decoratedItem.Update();
            }
        }
        
    }
    
}
